### ISSUE TITLE
Homebrew installation on new Mac fails when Tapping homebrew/cask 

### ISSUE THREAD

**Issue by ginderg** [Homebrew/homebrew-cask#100633]
#### General troubleshooting steps

- [ ] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [ ] I have retried my command with `--force`.
- [ ] I ran `brew update-reset && brew update` and retried my command.
- [ ] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [ ] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [ ] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/checksum_does_not_match_error.md) error.

#### Description of issue

Using a new Mac with M1 processor running macOS Big Sur 11.2.1
Made attempts at installing Hombrew but it fails at around 43% of receiving objects when running through the Tapping home-brew/core step.

1) Below is the entire information from my terminal window on the initial attempt:

~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password).
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew
==> The following new directories will be created:
/opt/homebrew/bin
/opt/homebrew/etc
/opt/homebrew/include
/opt/homebrew/lib
/opt/homebrew/sbin
/opt/homebrew/share
/opt/homebrew/var
/opt/homebrew/opt
/opt/homebrew/share/zsh
/opt/homebrew/share/zsh/site-functions
/opt/homebrew/var/homebrew
/opt/homebrew/var/homebrew/linked
/opt/homebrew/Cellar
/opt/homebrew/Caskroom
/opt/homebrew/Frameworks

Press RETURN to continue or any other key to abort
==> /usr/bin/sudo /bin/mkdir -p /opt/homebrew
==> /usr/bin/sudo /usr/sbin/chown root:wheel /opt/homebrew
==> /usr/bin/sudo /bin/mkdir -p /opt/homebrew/bin /opt/homebrew/etc /opt/homebrew/include /opt/homebrew/lib /opt/homebrew/sbin /opt/homebrew/share /opt/homebrew/var /opt/homebrew/opt /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions /opt/homebrew/var/homebrew /opt/homebrew/var/homebrew/linked /opt/homebrew/Cellar /opt/homebrew/Caskroom /opt/homebrew/Frameworks
==> /usr/bin/sudo /bin/chmod g+rwx /opt/homebrew/bin /opt/homebrew/etc /opt/homebrew/include /opt/homebrew/lib /opt/homebrew/sbin /opt/homebrew/share /opt/homebrew/var /opt/homebrew/opt /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions /opt/homebrew/var/homebrew /opt/homebrew/var/homebrew/linked /opt/homebrew/Cellar /opt/homebrew/Caskroom /opt/homebrew/Frameworks
==> /usr/bin/sudo /usr/sbin/chown gindergabri /opt/homebrew/bin /opt/homebrew/etc /opt/homebrew/include /opt/homebrew/lib /opt/homebrew/sbin /opt/homebrew/share /opt/homebrew/var /opt/homebrew/opt /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions /opt/homebrew/var/homebrew /opt/homebrew/var/homebrew/linked /opt/homebrew/Cellar /opt/homebrew/Caskroom /opt/homebrew/Frameworks
==> /usr/bin/sudo /usr/bin/chgrp admin /opt/homebrew/bin /opt/homebrew/etc /opt/homebrew/include /opt/homebrew/lib /opt/homebrew/sbin /opt/homebrew/share /opt/homebrew/var /opt/homebrew/opt /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions /opt/homebrew/var/homebrew /opt/homebrew/var/homebrew/linked /opt/homebrew/Cellar /opt/homebrew/Caskroom /opt/homebrew/Frameworks
==> /usr/bin/sudo /usr/sbin/chown -R gindergabri:admin /opt/homebrew
==> Downloading and installing Homebrew...
remote: Enumerating objects: 44, done.
remote: Counting objects: 100% (44/44), done.
remote: Compressing objects: 100% (44/44), done.
remote: Total 175898 (delta 0), reused 44 (delta 0), pack-reused 175854
Receiving objects: 100% (175898/175898), 45.79 MiB | 4.17 MiB/s, done.
Resolving deltas: 100% (130550/130550), done.
From https://github.com/Homebrew/brew
 * [new branch]          dependabot/bundler/Library/Homebrew/sorbet-0.5.6287 -> origin/dependabot/bundler/Library/Homebrew/sorbet-0.5.6287
 * [new branch]          master                                              -> origin/master
 * [new tag]             0.1                                                 -> 0.1
 * [new tag]             0.2                                                 -> 0.2
 * [new tag]             0.3                                                 -> 0.3
 * [new tag]             0.4                                                 -> 0.4
 * [new tag]             0.5                                                 -> 0.5
 * [new tag]             0.6                                                 -> 0.6
 * [new tag]             0.7                                                 -> 0.7
 * [new tag]             0.7.1                                               -> 0.7.1
 * [new tag]             0.8                                                 -> 0.8
 * [new tag]             0.8.1                                               -> 0.8.1
 * [new tag]             0.9                                                 -> 0.9
 * [new tag]             0.9.1                                               -> 0.9.1
 * [new tag]             0.9.2                                               -> 0.9.2
 * [new tag]             0.9.3                                               -> 0.9.3
 * [new tag]             0.9.4                                               -> 0.9.4
 * [new tag]             0.9.5                                               -> 0.9.5
 * [new tag]             0.9.8                                               -> 0.9.8
 * [new tag]             0.9.9                                               -> 0.9.9
 * [new tag]             1.0.0                                               -> 1.0.0
 * [new tag]             1.0.1                                               -> 1.0.1
 * [new tag]             1.0.2                                               -> 1.0.2
 * [new tag]             1.0.3                                               -> 1.0.3
 * [new tag]             1.0.4                                               -> 1.0.4
 * [new tag]             1.0.5                                               -> 1.0.5
 * [new tag]             1.0.6                                               -> 1.0.6
 * [new tag]             1.0.7                                               -> 1.0.7
 * [new tag]             1.0.8                                               -> 1.0.8
 * [new tag]             1.0.9                                               -> 1.0.9
 * [new tag]             1.1.0                                               -> 1.1.0
 * [new tag]             1.1.1                                               -> 1.1.1
 * [new tag]             1.1.10                                              -> 1.1.10
 * [new tag]             1.1.11                                              -> 1.1.11
 * [new tag]             1.1.12                                              -> 1.1.12
 * [new tag]             1.1.13                                              -> 1.1.13
 * [new tag]             1.1.2                                               -> 1.1.2
 * [new tag]             1.1.3                                               -> 1.1.3
 * [new tag]             1.1.4                                               -> 1.1.4
 * [new tag]             1.1.5                                               -> 1.1.5
 * [new tag]             1.1.6                                               -> 1.1.6
 * [new tag]             1.1.7                                               -> 1.1.7
 * [new tag]             1.1.8                                               -> 1.1.8
 * [new tag]             1.1.9                                               -> 1.1.9
 * [new tag]             1.2.0                                               -> 1.2.0
 * [new tag]             1.2.1                                               -> 1.2.1
 * [new tag]             1.2.2                                               -> 1.2.2
 * [new tag]             1.2.3                                               -> 1.2.3
 * [new tag]             1.2.4                                               -> 1.2.4
 * [new tag]             1.2.5                                               -> 1.2.5
 * [new tag]             1.2.6                                               -> 1.2.6
 * [new tag]             1.3.0                                               -> 1.3.0
 * [new tag]             1.3.1                                               -> 1.3.1
 * [new tag]             1.3.2                                               -> 1.3.2
 * [new tag]             1.3.3                                               -> 1.3.3
 * [new tag]             1.3.4                                               -> 1.3.4
 * [new tag]             1.3.5                                               -> 1.3.5
 * [new tag]             1.3.6                                               -> 1.3.6
 * [new tag]             1.3.7                                               -> 1.3.7
 * [new tag]             1.3.8                                               -> 1.3.8
 * [new tag]             1.3.9                                               -> 1.3.9
 * [new tag]             1.4.0                                               -> 1.4.0
 * [new tag]             1.4.1                                               -> 1.4.1
 * [new tag]             1.4.2                                               -> 1.4.2
 * [new tag]             1.4.3                                               -> 1.4.3
 * [new tag]             1.5.0                                               -> 1.5.0
 * [new tag]             1.5.1                                               -> 1.5.1
 * [new tag]             1.5.10                                              -> 1.5.10
 * [new tag]             1.5.11                                              -> 1.5.11
 * [new tag]             1.5.12                                              -> 1.5.12
 * [new tag]             1.5.13                                              -> 1.5.13
 * [new tag]             1.5.14                                              -> 1.5.14
 * [new tag]             1.5.2                                               -> 1.5.2
 * [new tag]             1.5.3                                               -> 1.5.3
 * [new tag]             1.5.4                                               -> 1.5.4
 * [new tag]             1.5.5                                               -> 1.5.5
 * [new tag]             1.5.6                                               -> 1.5.6
 * [new tag]             1.5.7                                               -> 1.5.7
 * [new tag]             1.5.8                                               -> 1.5.8
 * [new tag]             1.5.9                                               -> 1.5.9
 * [new tag]             1.6.0                                               -> 1.6.0
 * [new tag]             1.6.1                                               -> 1.6.1
 * [new tag]             1.6.10                                              -> 1.6.10
 * [new tag]             1.6.11                                              -> 1.6.11
 * [new tag]             1.6.12                                              -> 1.6.12
 * [new tag]             1.6.13                                              -> 1.6.13
 * [new tag]             1.6.14                                              -> 1.6.14
 * [new tag]             1.6.15                                              -> 1.6.15
 * [new tag]             1.6.16                                              -> 1.6.16
 * [new tag]             1.6.17                                              -> 1.6.17
 * [new tag]             1.6.2                                               -> 1.6.2
 * [new tag]             1.6.3                                               -> 1.6.3
 * [new tag]             1.6.4                                               -> 1.6.4
 * [new tag]             1.6.5                                               -> 1.6.5
 * [new tag]             1.6.6                                               -> 1.6.6
 * [new tag]             1.6.7                                               -> 1.6.7
 * [new tag]             1.6.8                                               -> 1.6.8
 * [new tag]             1.6.9                                               -> 1.6.9
 * [new tag]             1.7.0                                               -> 1.7.0
 * [new tag]             1.7.1                                               -> 1.7.1
 * [new tag]             1.7.2                                               -> 1.7.2
 * [new tag]             1.7.3                                               -> 1.7.3
 * [new tag]             1.7.4                                               -> 1.7.4
 * [new tag]             1.7.5                                               -> 1.7.5
 * [new tag]             1.7.6                                               -> 1.7.6
 * [new tag]             1.7.7                                               -> 1.7.7
 * [new tag]             1.8.0                                               -> 1.8.0
 * [new tag]             1.8.1                                               -> 1.8.1
 * [new tag]             1.8.2                                               -> 1.8.2
 * [new tag]             1.8.3                                               -> 1.8.3
 * [new tag]             1.8.4                                               -> 1.8.4
 * [new tag]             1.8.5                                               -> 1.8.5
 * [new tag]             1.8.6                                               -> 1.8.6
 * [new tag]             1.9.0                                               -> 1.9.0
 * [new tag]             1.9.1                                               -> 1.9.1
 * [new tag]             1.9.2                                               -> 1.9.2
 * [new tag]             1.9.3                                               -> 1.9.3
 * [new tag]             2.0.0                                               -> 2.0.0
 * [new tag]             2.0.1                                               -> 2.0.1
 * [new tag]             2.0.2                                               -> 2.0.2
 * [new tag]             2.0.3                                               -> 2.0.3
 * [new tag]             2.0.4                                               -> 2.0.4
 * [new tag]             2.0.5                                               -> 2.0.5
 * [new tag]             2.0.6                                               -> 2.0.6
 * [new tag]             2.1.0                                               -> 2.1.0
 * [new tag]             2.1.1                                               -> 2.1.1
 * [new tag]             2.1.10                                              -> 2.1.10
 * [new tag]             2.1.11                                              -> 2.1.11
 * [new tag]             2.1.12                                              -> 2.1.12
 * [new tag]             2.1.13                                              -> 2.1.13
 * [new tag]             2.1.14                                              -> 2.1.14
 * [new tag]             2.1.15                                              -> 2.1.15
 * [new tag]             2.1.16                                              -> 2.1.16
 * [new tag]             2.1.2                                               -> 2.1.2
 * [new tag]             2.1.3                                               -> 2.1.3
 * [new tag]             2.1.4                                               -> 2.1.4
 * [new tag]             2.1.5                                               -> 2.1.5
 * [new tag]             2.1.6                                               -> 2.1.6
 * [new tag]             2.1.7                                               -> 2.1.7
 * [new tag]             2.1.8                                               -> 2.1.8
 * [new tag]             2.1.9                                               -> 2.1.9
 * [new tag]             2.2.0                                               -> 2.2.0
 * [new tag]             2.2.1                                               -> 2.2.1
 * [new tag]             2.2.10                                              -> 2.2.10
 * [new tag]             2.2.11                                              -> 2.2.11
 * [new tag]             2.2.12                                              -> 2.2.12
 * [new tag]             2.2.13                                              -> 2.2.13
 * [new tag]             2.2.14                                              -> 2.2.14
 * [new tag]             2.2.15                                              -> 2.2.15
 * [new tag]             2.2.16                                              -> 2.2.16
 * [new tag]             2.2.17                                              -> 2.2.17
 * [new tag]             2.2.2                                               -> 2.2.2
 * [new tag]             2.2.3                                               -> 2.2.3
 * [new tag]             2.2.4                                               -> 2.2.4
 * [new tag]             2.2.5                                               -> 2.2.5
 * [new tag]             2.2.6                                               -> 2.2.6
 * [new tag]             2.2.7                                               -> 2.2.7
 * [new tag]             2.2.8                                               -> 2.2.8
 * [new tag]             2.2.9                                               -> 2.2.9
 * [new tag]             2.3.0                                               -> 2.3.0
 * [new tag]             2.4.0                                               -> 2.4.0
 * [new tag]             2.4.1                                               -> 2.4.1
 * [new tag]             2.4.10                                              -> 2.4.10
 * [new tag]             2.4.11                                              -> 2.4.11
 * [new tag]             2.4.12                                              -> 2.4.12
 * [new tag]             2.4.13                                              -> 2.4.13
 * [new tag]             2.4.14                                              -> 2.4.14
 * [new tag]             2.4.15                                              -> 2.4.15
 * [new tag]             2.4.16                                              -> 2.4.16
 * [new tag]             2.4.2                                               -> 2.4.2
 * [new tag]             2.4.3                                               -> 2.4.3
 * [new tag]             2.4.4                                               -> 2.4.4
 * [new tag]             2.4.5                                               -> 2.4.5
 * [new tag]             2.4.6                                               -> 2.4.6
 * [new tag]             2.4.7                                               -> 2.4.7
 * [new tag]             2.4.8                                               -> 2.4.8
 * [new tag]             2.4.9                                               -> 2.4.9
 * [new tag]             2.5.0                                               -> 2.5.0
 * [new tag]             2.5.1                                               -> 2.5.1
 * [new tag]             2.5.10                                              -> 2.5.10
 * [new tag]             2.5.11                                              -> 2.5.11
 * [new tag]             2.5.12                                              -> 2.5.12
 * [new tag]             2.5.2                                               -> 2.5.2
 * [new tag]             2.5.3                                               -> 2.5.3
 * [new tag]             2.5.4                                               -> 2.5.4
 * [new tag]             2.5.5                                               -> 2.5.5
 * [new tag]             2.5.6                                               -> 2.5.6
 * [new tag]             2.5.7                                               -> 2.5.7
 * [new tag]             2.5.8                                               -> 2.5.8
 * [new tag]             2.5.9                                               -> 2.5.9
 * [new tag]             2.6.0                                               -> 2.6.0
 * [new tag]             2.6.1                                               -> 2.6.1
 * [new tag]             2.6.2                                               -> 2.6.2
 * [new tag]             2.7.0                                               -> 2.7.0
 * [new tag]             2.7.1                                               -> 2.7.1
 * [new tag]             2.7.2                                               -> 2.7.2
 * [new tag]             2.7.3                                               -> 2.7.3
 * [new tag]             2.7.4                                               -> 2.7.4
 * [new tag]             2.7.5                                               -> 2.7.5
 * [new tag]             2.7.6                                               -> 2.7.6
 * [new tag]             2.7.7                                               -> 2.7.7
 * [new tag]             3.0.0                                               -> 3.0.0
 * [new tag]             3.0.1                                               -> 3.0.1
 * [new tag]             3.0.2                                               -> 3.0.2
 * [new tag]             3.0.3                                               -> 3.0.3
 * [new tag]             3.0.4                                               -> 3.0.4
HEAD is now at 9d62a13a3 Merge pull request #10700 from appropriate/docs-homebrew-and-java
==> Tapping homebrew/core
remote: Enumerating objects: 116, done.
remote: Counting objects: 100% (116/116), done.
remote: Compressing objects: 100% (78/78), done.
fatal: The remote end hung up unexpectedly22.19 MiB | 4.25 MiB/s   
fatal: early EOF
fatal: index-pack failed
Failed during: git fetch --force origin refs/heads/master:refs/remotes/origin/master

2) I then tried running this command and get the results below:

~ % brew update-reset && brew update
==> Fetching /opt/homebrew...

==> Resetting /opt/homebrew...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'

==> Fetching /opt/homebrew/Library/Taps/homebrew/homebrew-core...
remote: Enumerating objects: 116, done.
remote: Counting objects: 100% (116/116), done.
remote: Compressing objects: 100% (54/54), done.
fatal: the remote end hung up unexpectedly10.37 MiB | 3.74 MiB/s
fatal: early EOF
fatal: index-pack failed
error: Not a valid ref: refs/remotes/origin/master

==> Resetting /opt/homebrew/Library/Taps/homebrew/homebrew-core...
fatal: ref refs/remotes/origin/HEAD is not a symbolic ref
fatal: 'origin/HEAD' is not a commit and a branch '' cannot be created from it

error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
Error: Fetching /opt/homebrew/Library/Taps/homebrew/homebrew-core failed!
==> Homebrew has enabled anonymous aggregate formula and cask analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics have been recorded yet (nor will be during this `brew` run).

==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations

3) I then try to see if I can install core using this command and the results are below:

~ % brew install core
error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Updating Homebrew...
==> Tapping homebrew/cask
Cloning into '/opt/homebrew/Library/Taps/homebrew/homebrew-cask'...
remote: Enumerating objects: 4, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (4/4), done.
fatal: the remote end hung up unexpectedly21.67 MiB | 4.26 MiB/s
fatal: early EOF
fatal: index-pack failed
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-cask /opt/homebrew/Library/Taps/homebrew/homebrew-cask --origin=origin` exited with 128.

4) I then try to run the git clone command to see what happens:

 ~ % git clone https://github.com/Homebrew/homebrew-cask /opt/homebrew/Library/Taps/homebrew/homebrew-cask --origin=origin 
Cloning into '/opt/homebrew/Library/Taps/homebrew/homebrew-cask'...
remote: Enumerating objects: 4, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (4/4), done.
fatal: The remote end hung up unexpectedly23.01 MiB | 4.27 MiB/s   
fatal: early EOF
fatal: index-pack failed

#### Command that failed

```
{{replace this}}
```

#### Output of command with `--verbose --debug`

```
{{replace this}}
```

#### Output of `brew doctor --verbose`

```
{{replace this}}
```

#### Output of `brew tap`

```
{{replace this}}
```

**Comment by ginderg** [comment]
5 Results when I ran brew doctor

~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout

**Comment by ginderg** [comment]
6.~ % git -C $(brew --repo homebrew/core) checkout
fatal: You are on a branch yet to be born

**Comment by miccal** [comment]
If `brew update-reset` fails, then this seems to be an issue with your ability to connect to GitHub, and not with `homebrew-cask`.

Please open an issue at https://github.com/Homebrew/discussions to see if others can help you with this.

**Comment by ginderg** [comment]
Thanks, your input actually helped me figure this out, so leaving my comments here as it might help others.

The issue was definitely related to connection to GitHub via Terminal.  I had no issues browsing to GitHub on a web browser, however I decided to ping GitHub.com within a Terminal window, then I googled the IP address that was showing on the ping results, and it turned out not to be the GitHub IP address.  This pointed me to information that my ISP - Virgin Media's Web Safe settings are the cause of redirecting the requests, so I had to turn off my web safe settings, rebooted my router and opened a new terminal window and pinged github.com again, and this time I got a different IP address, so tried to install Homebrew again and this time it ran through all the steps without any errors.
